luigi/scheduler.py

```diff
@@ -591,6 +612,7 @@ def get_work(self, worker, host=None, **kwargs):
                               for worker in self._state.get_active_workers())
 
         tasks = list(self._state.get_pending_tasks())
+        self._state.cascade_disabled_state(tasks)
```

Comment:
Tasks go from disabled to pending after a time delay if you have automatic disabling enabled for repeated failures. Tasks also go from disabled to pending if you schedule a task as disabled and later schedule it as pending. It seems that the logic you'd actually want for this is to not count upstream disabled jobs as pending for the purposes of keep alive.